### ISSUE TITLE
environment variable `OUT_DIR` not defined at compile time

### ISSUE THREAD

**Issue by joepio** [static-files-rs/static-files#9]
Hi :) Thanks for making and sharing this crate.

I've been using this for a while to include my JS assets into [my project](https://github.com/atomicdata-dev/atomic-server/issues). Some time ago I decided to move the JS source code into my project, instead of copying the assets to my repo. I modified my `build.rs` to run `pnpm build`, which builds the assets. This works fine when building, but I haven't been able to `cargo publish` my repo.

That' because when `cargo publish` is run, the source code of the project is copied to the `/target` directory. Note that this does _not_ include adjacent / parent dirs. I have a monorepo with the JS source in one of the adjacent dirs. So when the [`build.rs`](https://github.com/atomicdata-dev/atomic-server/blob/2cb428075f12cd9ac0dd849e576e597c11a783c5/server/build.rs) tries to run `pnpm build` in the adjacent dir, it can't find it.

So I added steps that copy the directory... Which seems to work. Unfortunately, I'm not getting this error:

```
error: environment variable `OUT_DIR` not defined at compile time
  --> src/routes.rs:15:18
   |
15 | include!(concat!(env!("OUT_DIR"), "/generated.rs"));
   |                  ^^^^^^^^^^^^^^^
   |
   = help: Cargo sets build script variables at run time. Use `std::env::var("OUT_DIR")` instead
   = note: this error originates in the macro `env` (in Nightly builds, run with -Z macro-backtrace for more info)

error[E0425]: cannot find function `generate` in this scope
  --> src/routes.rs:25:37
   |
25 |             ResourceFiles::new("/", generate())
   |                                     ^^^^^^^^ not found in this scope
```

This ENV [should](https://doc.rust-lang.org/cargo/reference/environment-variables.html) be available during compilation.

I tried the suggestion, but using `std::env::var("OUT_DIR")` does not work, as `concat!` requires a literal.

Maybe someone reading this has an idea!

**Comment by kilork** [comment]
Hi, this looks like interesting case for study, because one of main goals of the project is to allow install of package with bundled static resources using `cargo install` command. I do not think I completely understand, but from there the error started to popup? `build.rs` should not run during without this `OUT_DIR` variable.

**Comment by kilork** [comment]
I am doing some similar things in this crate, not sure it is really relevant, but still worth a look: https://github.com/rust-sys-lib/libxml2

It is work in progress, things may not work as expected.

**Comment by joepio** [comment]
@kilork Thanks for responding!

The error appears _only_ when doing `cargo publish`, and it only started to appear after I changed how I included the assets. I went from "it's part of the committed files" to "I copy it during the `build.rs` process to the source folder".

- I have a monorepo that contains a folder for the Rust server + TS frontend 
- `build.rs` checks for a TS folder. If it finds one, and it finds changes in the TS folder (is any file changed after the `dist` files were built?) it tries to run `pnpm build`. Then, it clones the `dist` files to the server folder `assets_tmp`.
- When running `cargo publish` command, the `archive` step copies all the project dirs, which includes the `assets_tmp` folder. I can verify this is happening correctly.
- One thing that surprises me is that when I run `cargo publish`, is that I don't even see any evidence of any `build.rs` step actually happening. So the `OUT_DIR` env issue might appear _before_ all of the above is even relevant?
